src/OutputWriters/windowed_time_average.jl

```diff
@@ -83,18 +84,59 @@ JLD2OutputWriter scheduled on TimeInterval(4 days):
 """
 function AveragedTimeInterval(interval; window=interval, stride=1)
     window > interval && throw(ArgumentError("Averaging window $window is greater than the output interval $interval."))
-    return AveragedTimeInterval(Float64(interval), Float64(window), stride, 0.0, false)
+    return AveragedTimeInterval(Float64(interval), Float64(window), stride, 0.0, 0, false)
 end
 
+# function initialize_schedule!(sch::AveragedTimeInterval, first_actuation_time::Number)
+#     sch.first_actuation_time = first_actuation_time
+#     sch.actuations = 0
+#     return true
+# end
+
+# initialize_schedule!(sch::AveragedTimeInterval, model) = initialize_schedule!(sch, model.clock.time)
+
+
+function next_actuation_time(sch::AveragedTimeInterval)
+    t₀ = sch.first_actuation_time
+    N = sch.actuations
+    T = sch.interval
+    return t₀ + (N + 1) * T
+end
+
+function (sch::AveragedTimeInterval)(model)
+    t = model.clock.time
+    t★ = next_actuation_time(sch)
+
+    if t >= t★
+        if sch.actuations < typemax(Int)
+            sch.actuations += 1
+        else # re-initialize the schedule to t★
+            initialize!(sch, t★)
+        end
+        return true
+    else
+        return false
+    end
+end
```

Comment:
I don't think this function can be carried over from `TimeInterval` like this. 

We need this function to return `true` every time the model should be incrementing the time-average. If `window == interval`, that should mean every single timestep! Right now, I think it is only returning `true` at the very end of a window.

That said, we also want the `actuations+=1` to happen once per interval (at the end of the collecting window).

Reply:
I think the `sch.actuations += 1` should be removed from here and moved into the appropriate point in `advance_time_average!`

Reply:
I think this should actually be more similar to the previous implementation, something like:

```
(sch::AveragedTimeInterval)(model) = sch.collecting || model.clock.time >= next_actuation_time(sch) - sch.window
```

This comparison operation is basically just asking if we are in the current collecting window (between `next_actuation_time(sch) - sch.window` and `next_actuation_time(sch)`) or in the non-collecting part of the interval outside of the window (between `next_actuation_time(sch) - sch.interval` and `next_actuation_time(sch) - sch.window`).